analyses/molecular-subtyping-ATRT/01-ATRT-molecular-subtyping-data-prep.Rmd

```diff
@@ -0,0 +1,391 @@
+```
+
+### ssGSEA data
+
+```{r}
+# Calculate ssGSEA mean and sd
+ssGSEA_means <- rowMeans(ssGSEA, na.rm = TRUE)
+ssGSEA_sd <- apply(ssGSEA, 1, sd, na.rm = TRUE)
+
+# Subtract mean
+ssGSEA_zscored <- sweep(ssGSEA, 1, ssGSEA_means, FUN = "-")
+
+# Divide by SD remove NAs and Inf values from zscore for genes with 0 in normData
+ssGSEA_zscored <- sweep(ssGSEA_zscored, 1, ssGSEA_sd, FUN = "/") %>% 
+  dplyr::na_if(Inf) %>% 
+  na.omit()
+
+# Transpose
+transposed_ssGSEA <- t(ssGSEA_zscored)
+
+# Select wanted pathways and merge metadata 
+transposed_ssGSEA <- transposed_ssGSEA %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  dplyr::select(
+    Kids_First_Biospecimen_ID,
+    HALLMARK_MYC_TARGETS_V1,
+    HALLMARK_MYC_TARGETS_V2,
+    HALLMARK_NOTCH_SIGNALING
+  ) %>%
+  dplyr::left_join(select_metadata, by = "Kids_First_Biospecimen_ID")
+
+#Display `transposed_ssGSEA`
+transposed_ssGSEA %>%
+  head(n = 15) %>%
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+
+# Join ATRT expression and focal CN data with transposed ssGSEA data
+atrt_expression_cn_df <- atrt_expression_cn_df %>%
+  dplyr::left_join(transposed_ssGSEA,
+                    by = "sample_id")
+```
+
+### Tumor mutation burden data
+
+```{r}
+# Join tumor mutuation data with metadata
+tmb_df <- tmb_df %>%
+  dplyr::select(Tumor_Sample_Barcode, tmb) %>%
+  dplyr::inner_join(select_metadata,
+                   by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID"))
+
+#Display `tmb_df`
+tmb_df %>%
+  head(n = 15) %>%
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+
+# Join ATRT expression, focal CN data and transposed ssGSEA data with tumor 
+# mutation burden data
+atrt_expression_cn_tmb_df <- atrt_expression_cn_df %>%
+  dplyr::left_join(tmb_df, by = "sample_id")
+
+## TODO: Add a column to this data.frame denoting `chr22q` loss using the SV
+# data.
+```
+
+# Save results
+
+```{r}
+# Save final data.frame
+final_df <- atrt_expression_cn_tmb_df %>%
+  dplyr::group_by(sample_id) %>%
+  dplyr::summarise(Kids_First_Biospecimen_ID = paste(sort(unique(
+    Kids_First_Biospecimen_ID.x
+  )), collapse = ", "),
+  Kids_First_Participant_ID = paste(sort(unique(Kids_First_Participant_ID.x)), collapse = ", "),
+  location_summary = paste(sort(unique(location_summary)), collapse = ", "),
+  age_at_diagnosis_days = paste(sort(unique(age_at_diagnosis_days)), collapse = ", "),
+  germline_sex_estimate = paste(sort(unique(germline_sex_estimate)), collapse = ", "),
+  SMARCB1_focal_status = paste(sort(unique(SMARCB1_focal_status)), collapse = ", "),
+  SMARCA4_focal_status = paste(sort(unique(SMARCA4_focal_status)), collapse = ", "),
+  Overexpressed_gene_sets = paste(sort(unique(Overexpressed_gene_sets)), collapse = ", "),
+  HALLMARK_MYC_TARGETS_V1 = mean(HALLMARK_MYC_TARGETS_V1),
+  HALLMARK_MYC_TARGETS_V2 = mean(HALLMARK_MYC_TARGETS_V2),
+  HALLMARK_NOTCH_SIGNALLING = mean(HALLMARK_NOTCH_SIGNALING),
+  tmb = mean(tmb),
```

Comment:
I assume that you have to do all of these `mean` steps because you have duplicate sample identifiers. If that's correct, I'd try summarizing the CN data before joining it with the other data.